wormhole-connect/src/utils/routes/operator.ts

```diff
@@ -141,13 +137,90 @@ export default class Operator {
     );
   }
 
+  allSupportedChains(): ChainName[] {
+    const supported = new Set<ChainName>();
+    for (const key in CHAINS) {
+      const chainName = key as ChainName;
+      for (const route of ROUTES) {
+        if (!supported.has(chainName)) {
+          const isSupported = this.isSupportedChain(route as Route, chainName);
+          if (isSupported) {
+            supported.add(chainName);
+          }
+        }
+      }
+    }
+    return Array.from(supported);
+  }
+
+  async allSupportedSourceTokens(
+    destToken: TokenConfig | undefined,
+    sourceChain?: ChainName | ChainId,
+    destChain?: ChainName | ChainId,
+  ): Promise<TokenConfig[]> {
+    const supported: { [key: string]: TokenConfig } = {};
+    for (const route of ROUTES) {
+      for (const key in TOKENS) {
+        const alreadySupported = supported[key];
+        if (!alreadySupported) {
+          const isSupported = await this.isSupportedSourceToken(
+            route as Route,
+            TOKENS[key],
+            destToken,
+            sourceChain,
+            destChain,
+          );
+          if (isSupported) {
+            supported[key] = TOKENS[key];
+          }
+        }
+      }
+    }
+    return Object.values(supported);
+  }
+
+  async allSupportedDestTokens(
+    sourceToken: TokenConfig | undefined,
+    sourceChain?: ChainName | ChainId,
+    destChain?: ChainName | ChainId,
+  ): Promise<TokenConfig[]> {
+    const supported: { [key: string]: TokenConfig } = {};
+    for (const route of ROUTES) {
+      for (const key in TOKENS) {
+        const alreadySupported = supported[key];
+        if (!alreadySupported) {
```

Comment:
We were querying all tokens for every route.  Tried to cut that down a bit by checking if they were already on the list